### PR TITLE
Use integer for primary key in L_TRANSFERS

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

File: src/models/db/transfers.js [hunk 1]
```diff
@@ -69,42 +71,64 @@ function convertToPersistent (data) {
...
 }
 
 function * updateTransfer (transfer, options) {
-  return db.update(transfer, {TRANSFER_ID: transfer.id},
-    options && options.transaction).then((result) => {
-      return adjustments.upsertAdjustments(transfer, options).then(() => result)
+  const transaction = options && options.transaction
+  return db.update(transfer, {TRANSFER_UUID: transfer.id}, transaction)
+  .then((result) => {
+    return db.selectOne({TRANSFER_UUID: transfer.id}, transaction)
```
Comment (MatthewPhinney) [inline src/models/db/transfers.js]: So the issue here is that knex won't return the autogenerated ID on an update?

Comment (clark800) [inline src/models/db/transfers.js]: Yeah, `update` "Resolves the promise / fulfills the callback with the number of affected rows for the query."

File: src/models/db/transfers.js [hunk 2]
```diff
@@ -69,42 +71,64 @@ function convertToPersistent (data) {
...
     .then((transfer) => {
-      return adjustments.getAdjustments(id, options).then((adjustments) => {
+      if (transfer === null) {
+        return null
+      }
+      return adjustments.getAdjustments(transfer._id, options)
+      .then((adjustments) => {
         const result = _.assign({}, transfer, adjustments)
-        return _.isEmpty(result) ? null : result
+        return _.isEmpty(result) ? null : _.omit(result, '_id')
```
Comment (MatthewPhinney) [inline src/models/db/transfers.js]: Might be clearer to omit the `_id` in the `convertToExternal`.

**Comment by MatthewPhinney** [comment]
LGTM

File: src/models/db/transfers.js (hunk 2 above)
Comment (clark800) [inline src/models/db/transfers.js]: Then we wouldn't be able to use `selectOne`, which creates a lot more duplicated code

**Pull request merged**